src/actors/symbolication.rs

```diff
@@ -283,19 +284,36 @@ impl SymbolicationActor {
     {
         let (sender, receiver) = oneshot::channel();
 
+        let hub = Arc::new(sentry::Hub::new_from_top(sentry::Hub::current()));
+
         // Assume that there are no UUID4 collisions in practice.
         let requests = self.requests.clone();
         let request_id = RequestId::new(uuid::Uuid::new_v4());
         requests.lock().insert(request_id, receiver.shared());
+        let drop_hub = hub.clone();
         let token = CallOnDrop::new(move || {
             requests.lock().remove(&request_id);
+            // we consider every premature drop of the future as fatal crash
+            drop_hub.end_session_with_status(SessionStatus::Crashed);
```

Comment:
Is this a noop if the session has already ended before or do we need to gate against this?

Reply:
yes it is. ending a session consumes it from the current hub.

Reply:
Thanks! I think a comment for this would increase clarity.